app/scripts/lib/mnemonic.ts

```diff
@@ -0,0 +1,418 @@
+/* eslint-disable no-bitwise, no-plusplus */
```

Comment:
Hey, so this file is _a lot_ to grok. It's very low level, as it optimizes for both memory use and CPU time. You'll want to read the comments and JSDoc descriptions, and maybe have some AI handy to do some explaining.